scenarios/mongoose_docker_one_to_one.erl

```diff
@@ -26,108 +25,71 @@
     timer:sleep(5000),
 
     NeighbourIds = lists:delete(MyId, lists:seq(max(1,MyId-?NUMBER_OF_PREV_NEIGHBOURS),
                                                 MyId+?NUMBER_OF_NEXT_NEIGHBOURS)),
     send_messages_many_times(Client, ?SLEEP_TIME_AFTER_EVERY_MESSAGE, NeighbourIds);
 do(_Other, _MyId, Client) ->
     lager:info("checker"),
-    send_presence_available(Client),
+    amoc_xmpp:send_presence_available(Client),
     escalus_connection:wait_forever(Client).
 
--spec send_presence_available(escalus:client()) -> ok.
-send_presence_available(Client) ->
-    Pres = escalus_stanza:presence(<<"available">>),
-    escalus_connection:send(Client, Pres).
-
--spec send_presence_unavailable(escalus:client()) -> ok.
-send_presence_unavailable(Client) ->
-    Pres = escalus_stanza:presence(<<"unavailable">>),
-    escalus_connection:send(Client, Pres).
-
--spec send_messages_many_times(escalus:client(), timeout(), [binjid()]) -> ok.
+-spec send_messages_many_times(escalus:client(), timeout(), [amoc_scenario:user_id()]) -> ok.
 send_messages_many_times(Client, MessageInterval, NeighbourIds) ->
     S = fun(_) ->
                 send_messages_to_neighbors(Client, NeighbourIds, MessageInterval)
         end,
     lists:foreach(S, lists:seq(1, ?NUMBER_OF_SEND_MESSAGE_REPEATS)).
 
 
--spec send_messages_to_neighbors(escalus:client(), [binjid()], timeout()) -> list().
+-spec send_messages_to_neighbors(escalus:client(), [amoc_scenario:user_id()], timeout()) -> list().
 send_messages_to_neighbors(Client, TargetIds, SleepTime) ->
-    [send_message(Client, make_jid(TargetId), SleepTime)
+    [send_message(Client, TargetId, SleepTime)
      || TargetId <- TargetIds].
 
--spec send_message(escalus:client(), binjid(), timeout()) -> ok.
+-spec send_message(escalus:client(), amoc_scenario:user_id(), timeout()) -> ok.
 send_message(Client, ToId, SleepTime) ->
-    MsgIn = make_message(ToId),
-    TimeStamp = integer_to_binary(usec:from_now(os:timestamp())),
-    escalus_connection:send(Client, escalus_stanza:setattr(MsgIn, <<"timestamp">>, TimeStamp)),
+    MsgIn = amoc_xmpp_stanzas:make_message_with_timestamp(ToId),
+    escalus_connection:send(Client, MsgIn),
     timer:sleep(SleepTime).
 
--spec make_message(binjid()) -> exml:element().
-make_message(ToId) ->
-    Body = <<"hello sir, you are a gentelman and a scholar.">>,
-    Id = escalus_stanza:id(),
-    escalus_stanza:set_id(escalus_stanza:chat_to(ToId, Body), Id).
-
--spec make_jid(amoc_scenario:user_id()) -> binjid().
-make_jid(Id) ->
-    BinInt = integer_to_binary(Id),
-    ProfileId = <<"user_", BinInt/binary>>,
-    Host = ?HOST,
-    << ProfileId/binary, "@", Host/binary >>.
-
--spec pick_server() -> binary().
-pick_server() ->
-    Servers = amoc_config:get(xmpp_servers),
-    S = size(Servers),
-    N = erlang:phash2(self(), S) + 1,
-    element(N, Servers).
+-spec send_and_recv_escalus_handlers() -> [{atom(), any()}].
+send_and_recv_escalus_handlers() ->
```

Comment:
I might be wrong but `send_and_recv_escalus_handlers` is used 4 times in different tests.
How about extracting this function too?